tests/workload_generator/corpus_test.py

```diff
@@ -22,78 +22,88 @@
-        "field2": "things"
-    }
-    doc_data = serialize_doc(doc)
-    client.count.return_value = {
-        "count": 1001
-    }
-    client.search.return_value = {
-        "_scroll_id": "uohialjrknf",
-        "_shards": {
-            "successful": 1,
-            "total": 1,
-            "skipped": 0
-        },
-        "hits": {
-            "hits": [
-                {
-                    "_index": "test",
-                    "_id": "0",
-                    "_score": 0,
-                    "_source": doc
-                }
-            ]
+    def serialize_doc(self, doc):
+        return (json.dumps(doc, separators=(",", ":")) + "\n").encode("utf-8")
+
+    @mock.patch("builtins.open", new_callable=mock.mock_open)
+    def test_extract(self, mo):
+        doc = {
+            "field1": "stuff",
+            "field2": "things"
+        }
+        doc_data = self.serialize_doc(doc)
+        self.mock_client.count.return_value = {
+            "count": 1001
+        }
+        self.mock_client.search.return_value = {
+            "_scroll_id": "uohialjrknf",
+            "_shards": {
+                "successful": 1,
+                "total": 1,
+                "skipped": 0
+            },
+            "hits": {
+                "hits": [
+                    {
+                        "_index": "test",
+                        "_id": "0",
+                        "_score": 0,
+                        "_source": doc
+                    }
+                ]
+            }
         }
-    }
 
-    def set_corp_size(*args, **kwargs):
-        path = args[0]
-        mockstat = mock.Mock()
-        if ".bz2" in path:
-            mockstat.st_size = 500
-        else:
-            mockstat.st_size = 1000
-        return mockstat
+        def set_corp_size(*args, **kwargs):
+            path = args[0]
+            mockstat = mock.Mock()
+            if ".bz2" in path:
+                mockstat.st_size = 500
+            else:
+                mockstat.st_size = 1000
+            return mockstat
 
-    client.scroll.return_value = {}
+        self.mock_client.scroll.return_value = {}
 
-    index = "test"
-    outdir = "/abs/outpath/to/workloads/"
+        index = "test"
 
-    with mock.patch("os.stat") as osstat:
-        osstat.side_effect = set_corp_size
-        res = corpus.extract(client, outdir, index)
-    assert mo.call_count == 4
-    mo.assert_has_calls([call("/abs/outpath/to/workloads/test-documents.json", "wb"),
-                         call("/abs/outpath/to/workloads/test-documents.json.bz2", "wb"),
-                         call("/abs/outpath/to/workloads/test-documents-1k.json", "wb"),
-                         call("/abs/outpath/to/workloads/test-documents-1k.json.bz2", "wb")
-                         ], any_order=True)
+        with mock.patch("os.stat") as osstat:
+            osstat.side_effect = set_corp_size
+            res = self.corpus_extractor.extract_documents(index)
+        assert mo.call_count == 4
```

Comment:
The `has_calls()` invocation will ensure there are at least 4 calls in any case, so this may be redundant -- unless you are concerned about repeated calls.

Reply:
This is to ensure that four types of files are present (the original corpora, the test mode version, and their two respective compressed versions).
```
        mo.assert_has_calls([call("/abs/outpath/to/workloads/test-documents.json", "wb"),
                            call("/abs/outpath/to/workloads/test-documents.json.bz2", "wb"),
                            call("/abs/outpath/to/workloads/test-documents-1k.json", "wb"),
                            call("/abs/outpath/to/workloads/test-documents-1k.json.bz2", "wb")
                            ], any_order=True)
```